blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -210,7 +209,58 @@ public static String getRecursivePathFromFullName(BluePipeline pipeline){
 
     @Override
     public Container<Resource> getActivities() {
-        return Containers.fromResource(getLink(), Lists.newArrayList(Iterators.concat(getQueue().iterator(), getRuns().iterator())));
+
+        return new Container<Resource>(){
+            @Override
+            public Iterator<Resource> iterator() {
+                return Lists.newArrayList(Iterators.concat(getQueue().iterator(), getRuns().iterator())).iterator();
```

Comment:
I also had a solution for this we talked about before... was just getting  back to that PR today, actually. My solution avoided reading all runs if the iterator wasn't read fully.